fastfield_codecs/src/monotonic_mapping.rs

```diff
@@ -34,6 +37,24 @@ pub trait StrictlyMonotonicFn<External, Internal> {
     fn mapping(&self, inp: External) -> Internal;
     /// Inverse of `mapping`. Maps the value from Internal to External.
     fn inverse(&self, out: Internal) -> External;
+
+    /// Maps a user provded value from External to Internal.
+    /// It may be necessary to coerce the value if it is outside the value space.
+    /// In that case it tries to find the next greater value in the value space.
+    ///
+    /// Returns a bool to mark if a value was outside the value space and had to be coerced _up_.
+    /// With that information we can detect if two values in a range both map outside the same value
+    /// space.
+    ///
+    /// coerce_up means the next valid upper value in the value space will be chosen if the value
+    /// has to be coerced.
+    fn mapping_coerce(&self, inp: External, _coerce_up: bool) -> (bool, Internal) {
```

Comment:
Would something like `.map_lower_than(&self, bound: Bound<To>) -> Bound<From>`  do the job?

If it would, I feel like the contract is more obvious.

Reply:
We could pass a Range in `mapping_coerce` instead (we don't have bounds here anymore). We still need the early exit for correctness if both map in front of the value space, and for that return a 1..0 range.